src/tests/build.sh

```diff
@@ -268,156 +70,70 @@ build_Tests()
-                echo "${__ErrMsgPrefix}${__MsgPrefix}Error: Check Test Build failed."
-                exit 1
-            fi
-        fi
-
-        echo "Managed tests build success!"
-
-        build_test_wrappers
-    fi
-
-    if [[ "$__CopyNativeTestBinaries" == 1 ]]; then
-        echo "Copying native test binaries to output..."
-
-        build_MSBuild_projects "Tests_Managed" "$__RepoRootDir/src/tests/build.proj" "Managed tests build (build tests)" "/p:RuntimeFlavor=$__RuntimeFlavor" "/t:CopyAllNativeProjectReferenceBinaries" "/bl:${__RepoRootDir}/artifacts/log/${__BuildType}/copy_native_test_binaries${__RuntimeFlavor}.binlog"
-
-        if [[ "$?" -ne 0 ]]; then
-            echo "${__ErrMsgPrefix}${__MsgPrefix}Error: copying native test binaries failed. Refer to the build log files for details (above)"
-            exit 1
-        fi
-    fi
-
-    if [[ -n "$__UpdateInvalidPackagesArg" ]]; then
-        __up="/t:UpdateInvalidPackageVersions"
-    fi
-
-    if [[ "$__SkipGenerateLayout" != 1 ]]; then
-        generate_layout
-    fi
-}
-
-build_MSBuild_projects()
-{
-    subDirectoryName="$1"
-    shift
-    projectName="$1"
-    shift
-    stepName="$1"
-    shift
-    extraBuildParameters=("$@")
-
     # Set up directories and file names
-    __BuildLogRootName="$subDirectoryName"
     __BuildLog="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.log"
     __BuildWrn="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.wrn"
     __BuildErr="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.err"
+    __BuildBinLog="$__LogsDir/${__BuildLogRootName}.${__TargetOS}.${__BuildArch}.${__BuildType}.binlog"
+    __msbuildLog="\"/flp:Verbosity=normal;LogFile=${__BuildLog}\""
+    __msbuildWrn="\"/flp1:WarningsOnly;LogFile=${__BuildWrn}\""
+    __msbuildErr="\"/flp2:ErrorsOnly;LogFile=${__BuildErr}\""
+
+    # Uncomment the line below when instrumenting Linux builds to produce binlogs;
+    # they seem to be too lengthy to be enabled by default.
+    __msbuildBinLog="\"/bl:${__BuildBinLog}\""
+
+    # Export properties as environment variables for the MSBuild scripts to use
+    export __TestDir
+    export __TestIntermediatesDir
+    export __NativeTestIntermediatesDir
+    export __BinDir
+    export __TestBinDir
+    export __SkipManaged
+    export __SkipGenerateLayout
+    export __SkipTestWrappers
+    export __BuildTestProject
+    export __BuildTestDir
+    export __BuildTestTree
+    export __RuntimeFlavor
+    export __CopyNativeProjectsAfterCombinedTestBuild
+    export __CopyNativeTestBinaries
+    export __Priority
+    export __DoCrossgen2
+    export __CreatePerfMap
+    export __CompositeBuildMode
+    export __BuildTestWrappersOnly
+    export __GenerateLayoutOnly
+    export __TestBuildMode
+    export __MonoAot
+    export __MonoFullAot
+    export __MonoBinDir
+    export __MsgPrefix
+    export __ErrMsgPrefix
+    export __Exclude
```

Comment:
I'm personally not a fan of how MSBuild implicitly imports the current environment as properties and I generally try to avoid it if possible. Additionally, if we ever are able to move to unifying the runtime test build script entry-point into the root build scripts, this will definitely need to be cleaned up for that. This can wait until later work though as the environment variables are easily identifiable.

Reply:
Yes, if you don't mind, I'd prefer to leave that to a separate follow-up change. The main problem is that escaping will be necessary at least for some of the properties (in particular I hit this with the <code>__MsgPrefix</code> and <code>__ErrMsgPrefix</code> variables) and that will require a bit more experimentation; moreover the escaping will be probably done in slightly different ways in the cmd/sh version of the script.